sdks/java/core/src/main/java/org/apache/beam/sdk/transforms/windowing/InvalidWindows.java

```diff
@@ -87,4 +87,9 @@ public void verifyCompatibility(WindowFn<?, ?> other) throws IncompatibleWindowE
   public Instant getOutputTime(Instant inputTimestamp, W window) {
     return inputTimestamp;
   }
+
+  @Override
+  public boolean isNonMerging() {
```

Comment:
It might be even better to change to `extends NonMergingWindowFn`.

I also think: wow! we chose the wrong default value for this

Reply:
+1